src/state/products.js

```diff
@@ -39,6 +39,39 @@ export const extractedProductShape = pt.shape({
   date: pt.string.isRequired,
 });
 
+/**
+ * Validate the contents of the given extracted product. This is used in lieu
+ * of proptypes since proptype checks do not work in a production build.
+ * @param {ExtractedProduct} extractedProduct
+ * @return {Boolean}
+ */
+export function isValidExtractedProduct(extractedProduct) {
+  if (typeof extractedProduct.title !== 'string') {
+    return false;
+  }
+
+  if (Number.isNaN(Date.parse(extractedProduct.date))) {
```

Comment:
According to `extractedProductShape`, `date` is of type `string`. Why are you parsing it into a number here?

Reply:
To confirm that it is a correctly-formatted date; `Date.parse` returns `NaN` if the date cannot be parsed.